### PR TITLE
chore: await_first_trial ensures first trial is returned

### PR DESCRIPTION
**Pull request by wes-turner**
Two changes, smooshed into one PR:

* Addition of unit test for ExperimentRef.await_first_trial.
  - To make this work, also added fixture for trials.

* Explicit ordering of trials requested from master so that when trials are returned the first trial is the one guaranteed to be returned.

[MLG-586]

## Test Plan
For an existing experiment, the following returns a trial if one exists. If none exist yet, the method waits until one does.
```python
from determined.experimental import client
client.get_experiment(<id>).await_first_trial()
```

## Checklist

- [ ] Changes have been manually QA'd
- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
  See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.

## Ticket

[MLG-586]: https://hpe-aiatscale.atlassian.net/browse/MLG-586?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: harness/determined/common/experimental/experiment.py [hunk 1]
```diff
@@ -116,7 +116,12 @@ def await_first_trial(self, interval: float = 0.1) -> trial.TrialReference:
         Wait for the first trial to be started for this experiment.
         """
         while True:
-            resp = bindings.get_GetExperimentTrials(self._session, experimentId=self._id)
+            resp = bindings.get_GetExperimentTrials(
+                self._session,
+                experimentId=self._id,
+                orderBy=bindings.v1OrderBy.ASC,
```
Comment (azhou-determined) [inline harness/determined/common/experimental/experiment.py:122]: what's the reasoning for the added order/sort parameters?

**Review by azhou-determined — APPROVED** [review]
lgtm

File: harness/determined/common/experimental/experiment.py (hunk 1 above)
Comment (wes-turner) [reply harness/determined/common/experimental/experiment.py:122]: To ensure that the method's functionality matched its docstring and name (or at least matched my interpretation of the docstring and name).

Comment (wes-turner) [reply harness/determined/common/experimental/experiment.py:122]: Should I undo?

**Pull request merged**